packages/node/src/methods/app-instance/take-action/operation.ts

```diff
@@ -0,0 +1,41 @@
+import { AppInstance } from "@counterfactual/machine";
+import { AppState } from "@counterfactual/types";
+import { Contract } from "ethers";
+import { Provider } from "ethers/providers";
+
+import { ERRORS } from "../../errors";
+
+export async function generateNewAppInstanceState(
+  appInstance: AppInstance,
+  action: any,
```

Comment:
We can define a new type alias here I think. `type AppAction = AppState`.

However at the same time I recommend renaming `AppState` to `ABIEncoderV2Struct` and reusing that here too.

Reply:
Sounds good. I'll make a follow up PR for that.